app/Models/BookRepository.php

```
@@ -1038,38 +1057,122 @@ private function stringInput(mixed $value): string
             }
         }
+
+        // Insert-first is deliberately non-destructive: if an id is invalid or
+        // this install has not applied the colorista enum migration yet, execute
+        // throws while every old link is still intact.
+        if ($desired !== []) {
+            $insert = $this->db->prepare(
+                'INSERT IGNORE INTO libri_autori (libro_id, autore_id, ruolo) VALUES (?, ?, ?)'
+            );
+            if ($insert === false) {
+                throw new \RuntimeException('Database error: unable to prepare contributor insert');
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '1124', column '27'). (undefined)

(MissingImport)